File: javascript/Car/car.test.js

const {startCar,stopCar,setFuelTank,getFuelTank,driveCar} = require("./car"); 

describe("Car Functions", () => {
  test("car can start", () => {
    let carStatus = false;
    carStatus = startCar(carStatus);
    expect(carStatus).toBe(true);
  });

  test("car can be turned off", () => {
    let carStatus = true;
    carStatus = stopCar(carStatus);
    expect(carStatus).toBe(false);
  });

  test("fuel tank can be set within capacity", () => {
    let fuelTank = 0;
    fuelTank = setFuelTank(fuelTank, 20);
    expect(fuelTank).toBe(20);
  });

  test("fuel tank cannot exceed capacity", () => {
    let fuelTank = 0;
    fuelTank = setFuelTank(fuelTank, 60); 
    expect(fuelTank).toBe(0); 
  });

  test("get fuel tank returns current value", () => {
    let fuelTank = 15;
    expect(getFuelTank(fuelTank)).toBe(15);
  });

  test("drive car consumes fuel correctly", () => {
    const distance = 40;
    const fuelConsumed = driveCar(distance, 8); 
    expect(fuelConsumed).toBe(5);
  });
});

File: javascript/Car/car_test.py
import unittest

from car import start_car, stop_car, set_fuel_tank, get_fuel_tank, drive_car


class TestCarFunctions(unittest.TestCase):
    def test_start_car(self):
        car_status = False
        car_status = start_car(car_status)
        self.assertTrue(car_status)

    def test_stop_car(self):
        car_status = True
        car_status = stop_car(car_status)
        self.assertFalse(car_status)

    def test_set_fuel_tank_within_capacity(self):
        fuel_tank = 0
        fuel_tank = set_fuel_tank(fuel_tank, 20)  # 20 liters
        self.assertEqual(fuel_tank, 20)

    def test_set_fuel_tank_exceeds_capacity(self):
        fuel_tank = 0
        fuel_tank = set_fuel_tank(fuel_tank, 60)  # exceeds 50 liters
        self.assertEqual(fuel_tank, 0)  # should remain unchanged

    def test_get_fuel_tank(self):
        fuel_tank = 15
        self.assertEqual(get_fuel_tank(fuel_tank), 15)

    def test_drive_car_consumption(self):
        distance = 40
        fuel_consumed = drive_car(distance, fuel_efficiency=8)  # 40/8 = 5
        self.assertEqual(fuel_consumed, 5)

